src/richie/apps/courses/templatetags/extra_tags.py

```diff
@@ -322,6 +324,46 @@ def joanie_product_widget_props(context):
 
     return json.dumps({"productId": product_id, "courseCode": course_code})
 
+@register.simple_tag(takes_context=True)
+def course_runs_list_widget_props(context):
+    """
+    Return a json dumps which contains all properties required
+    by CourseRunsList React widget.
+    """
+    request = context.get("request")
+    toolbar = get_toolbar_from_request(request)
+    edit = toolbar.edit_mode_active
+    course = context["course"]
+
+    queryset = course.course_runs
+
+    if not edit:
+        # Except if we are in edit mode,
+        # we don't want to show "hidden" and "to be scheduled" course runs
+        queryset = queryset.exclude(
+            Q(catalog_visibility=CourseRunCatalogVisibility.HIDDEN)
+            | Q(start__isnull=True)
+        )
+
+    course_runs = CourseRunSerializer(queryset, many=True).data
+
+    # This part was causing an error "CourseState" is not serializable"
```

Comment:
@jbpenrath mind this modification I had to do.

Reply:
Yup, that's why I pop the state property from data.